shopify_python/google_styleguide.py

```diff
@@ -0,0 +1,158 @@
+import sys
+
+import astroid  # pylint: disable=unused-import
+import six
+
+from pylint import checkers
+from pylint import interfaces
+from pylint import lint  # pylint: disable=unused-import
+
+
+if sys.version_info >= (3, 4):
+    import importlib.util  # pylint: disable=no-name-in-module,import-error
+else:
+    import importlib
+
+
+def register_checkers(linter):  # type: (lint.PyLinter) -> None
+    """Register checkers."""
+    linter.register_checker(GoogleStyleGuideChecker(linter))
+
+
+class GoogleStyleGuideChecker(checkers.BaseChecker):
+    """
+    Pylint checker for the Google Python Style Guide.
+
+    See https://google.github.io/styleguide/pyguide.html
+
+    Checks that can't be implemented include:
+      - When capturing an exception, use as rather than a comma
+    """
+    __implements__ = (interfaces.IAstroidChecker,)
+
+    name = 'google-styleguide-checker'
+
+    msgs = {
+        'C2601': ('Imported an object that is not a package or module',
+                  'import-modules-only',
+                  'Use imports for packages and modules only'),
+        'C2602': ('Imported using a partial path',
+                  'import-full-path',
+                  'Import each module using the full pathname location of the module.'),
+        'C2603': ('Variable declared at the module level (i.e. global)',
+                  'global-variable',
+                  'Avoid global variables in favor of class variables'),
+        'C2604': ('Raised two-argument exception',
+                  'two-arg-exception',
+                  "Use either raise Exception('message') or raise Exception"),
+        'C2605': ('Raised deprecated string-exception',
+                  'string-exception',
+                  "Use either raise Exception('message') or raise Exception"),
+        'C2606': ('Caught StandardError',
+                  'catch-standard-error',
+                  "Don't catch broad exceptions"),
+        'C2607': ('Try body too long',
+                  'try-too-long',
+                  "The larger the try body, the more likely that an unexpected exception will be raised"),
+        'C2608': ('Except body too long',
+                  'except-too-long',
+                  "The larger the except body, the more likely that an exception will be raised"),
+        'C2609': ('Finally body too long',
+                  'finally-too-long',
+                  "The larger the except body, the more likely that an exception will be raised"),
+    }
+
+    max_try_exc_finally_body_size = 5
```

Comment:
> If people intentionally rewrite their code to get around linter rules instead of adhering to the principle, that's what code review is for.

Code review is also for pointing out complexity problems like the ones this linter rule is meant to address. It's easier to spot too many lines of code in a single `try` block than too many lines of code split up into functions.

I'm in favour of the spirit of the rule, but not this particular implementation.

> Another option is to specify an overridable default, similar to the "max lines per module" rule.

I'd be in favour of something like this. Number of lines might even make the best metric since they're easy to understand (and therefore adjust) by other teams on other projects.